File: services/groupbuy-service/src/main/java/groupbuy_service/global/ErrorResponse.java
package groupbuy_service.global;

import lombok.Builder;
import lombok.Getter;

@Getter
public class ErrorResponse {

    private final int status;
    private final String msg;

    @Builder
    public ErrorResponse(int status, String msg) {
        this.status = status;
        this.msg = msg;
    }

    public static ErrorResponse from (ErrorCode errorCode) {
        return ErrorResponse.builder()
                .status(errorCode.getStatus())
                .msg(errorCode.getMsg())
                .build();
    }
}

File: services/inventory-service/src/main/kotlin/inventory_service/global/error/ErrorResponse.kt
package inventory_service.global.error

data class ErrorResponse(
    val status: Int,
    val message: String,
    val code: String
) {
    companion object {
        fun of(errorCode: ErrorCode): ErrorResponse {
            return ErrorResponse(
                status = errorCode.status,
                message = errorCode.msg,
                code = errorCode.name
            )
        }
    }
}
